### PR TITLE
Check in 680.py.

### PR DESCRIPTION
**Pull request by bunnygod**

### CODE REVIEW

File: 680.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                start += 1
+                end -= 1
+
+            return True
+
+        start, end = 0, len(s) - 1
+
+        while start < end:
+            if s[start] != s[end]:
+                return isPalindrome(s[start + 1 : end + 1]) or \
```
Comment (mujingw) [inline 680.py:19]: This part is hard to come up with. I was not able to solve this problem by myself early last month.

File: 680.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Solution:
+    def validPalindrome(self, s: str) -> bool:
+        def isPalindrome(s: str) -> bool:
```
[the remark this answers is not in the document]
Comment (bunnygod) [reply 680.py:3]: I can understand `continue` and `break`, but why `return`?  In this routine, we want to early return on failure case, right?

[the remark this answers is not in the document]
Comment (mujingw) [reply 680.py:3]: This structure:

```python
while start < end:
    if s[start] != s[end]:
        return False
```

given that the `if` block is at the very beginning of the loop body, is equivalent to:
```python
while start < end and s[start] == s[end]:
    ...
    ...
```

Both have early exit, but which one is more readable?

[the remark this answers is not in the document]
Comment (bunnygod) [reply 680.py:3]: But how do you code the return condition outside this `while` loop?  You need to do

```
return start >= end
```

This is more confusing to me at a glance.

[the remark this answers is not in the document]
Comment (mujingw) [reply 680.py:3]: `return start >= end` has a very clear meaning. It tells you the two pointers have validated all pairs of characters and finally converged in the middle.

**Pull request merged**